src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -996,9 +996,9 @@ public static void cacheRelease(Tangle tangle) throws Exception {
             Indexable hash = releaseQueue.poll();
             if (hash != null) {
                 TransactionViewModel tvm = cache.get(hash);
+                hashesToRelease.add(hash);
```

Comment:
Moving this makes the total transaction count increase. The scans for the `LatestMilestoneTracker` trigger the node to resave transactions in the cache. The previous commit where this was inside the below wrapping didn't have this problem.

Reply:
Moving that line outside is actually to align the cache size with the specified or default max configuration. Else, the cache grows past is allowed max size.
We should release all entries that should be released. Not only fresh entries.

Reply:
@DyrellC I don't think I understood you, will make sure to follow up

Reply:
@acha-bill moving it outside still has the unintended consequence of causing this to happen on the node after the pruning is done: 
```
01/22 03:07:25.266 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 32243
01/22 03:07:35.277 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 30025
01/22 03:07:45.285 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27775
01/22 03:07:55.286 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27784
01/22 03:08:05.289 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27792
01/22 03:08:15.293 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27801
01/22 03:08:25.296 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27810
01/22 03:08:35.302 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27818
01/22 03:08:45.305 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27827
01/22 03:08:55.309 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27836
01/22 03:09:05.313 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27844
01/22 03:09:15.321 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27853
01/22 03:09:25.325 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27861
01/22 03:09:35.329 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27870
01/22 03:09:45.330 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27880
```

Reply:
@GalRogozinski @acha-bill as you can see in the logs from the node, and if you run the test locally, after the node prunes the transactions, the total transaction count keeps rising. `CachePut` is called each time fromHash is called in `TransactionViewModel` and each time it is storing batches of transactions without releasing them from the cache release queue properly. So we're getting unintended to the db from already processed cache objects. This can be resolved by having the cache remove objects that are not in strongstore from the releasequeue instead of just returning null. I'll leave a comment where this fix can be applied

Reply:
:clap:

Reply:
Nice catch @DyrellC.
Also, to stop `cachePut` every time `fromHash` is called, we could make `fillMetaData` return boolean.
My initial thought was that if metadata updates, then we should update the cache.